pkg/server/backgroundsvcs/background_services.go

```diff
@@ -73,7 +73,15 @@ type BackgroundServiceRegistry struct {
 }
 
 func NewBackgroundServiceRegistry(services ...registry.BackgroundService) *BackgroundServiceRegistry {
-	return &BackgroundServiceRegistry{services}
+	svc := make([]registry.BackgroundService, 0, len(services))
```

Comment:
I'm not convinced about this change, can we instead unregister the service so its not in `services`?

Reply:
Not sure it is possible. 
1. The background service registry is created by wire that calls all required `ProvideService` and then injects the, into the registry. 
2. Then the list of services is copied downstream, and it will be pretty hard to try to remove all references to that service. 

Another way is to check for nil there 
https://github.com/grafana/grafana/blob/8bd825e16ca99baccd83d8325b02fe40c7640d43/pkg/server/server.go#L144-L143
but I think removing nils on providers side is more correct approach